HW2/P2/parallel_vector.pyx

```diff
@@ -81,10 +80,15 @@ cpdef move_data_fine_grained(np.int32_t[:] counts,
    ##########
    with nogil:
        for r in range(repeat):
-           for idx in range(src.shape[0]):
+           for idx in prange(src.shape[0], num_threads=4):  
                if counts[src[idx]] > 0:
```

Comment:
You could have optimized the code slightly by storing src[idx] and dest[idx] in local variables rather than looking them up every time.